sw/airborne/modules/computer_vision/viewvideo.c

```diff
@@ -109,23 +111,41 @@ struct viewvideo_t viewvideo = {
 #endif
 };
 
+struct image_t img_copy;
+
 /**
  * Handles all the video streaming and saving of the image shots
  * This is a sepereate thread, so it needs to be thread safe!
  */
 struct image_t *viewvideo_function(struct image_t *img);
-struct image_t *viewvideo_function(struct image_t *img)
-{
-  // Resize image if needed
-  struct image_t img_small;
-  image_create(&img_small,
-               img->w / viewvideo.downsize_factor,
-               img->h / viewvideo.downsize_factor,
-               IMAGE_YUV422);
+struct image_t *viewvideo_function(struct image_t *img) {
+  if (!viewvideo.new_image) {
+    // Free previous image
+    image_free(&img_copy);
```

Comment:
First time this breaks? And why create a new image every time? Since you can reuse the previous one, which saves a lot of malloc's and free's over time(and also a lot of processing time).